runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -111,3 +112,57 @@ def generate_cutoffs(df: pd.DataFrame, horizon: int, unit: str,
 
 def is_quaterly_alias(freq: str):
     return freq in QUATERLY_OFFSET_ALIAS
+
+def is_frequency_consistency(
+                start_time: Union[pd.Series, pd.Timestamp],
+                end_time: Union[pd.Series, pd.Timestamp], 
```

Comment:
Why do we need start_time can be both a pd.Series and a pd.Timestamp? Similarly for end_time. All the use cases above are passing a timestamp as start_time and a series of timestamps as end_time. If it's the only case, let's specify the exact type and maybe rename end_time to something like ds_series.